catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/UiNFCTestUtils.java

```diff
@@ -0,0 +1,147 @@
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+
+public final class UiNFCTestUtils {
+	private UiNFCTestUtils() {
+		throw new AssertionError();
+	}
+
+	public static final String FIRST_TEST_TAG_ID = "111111";
+	public static final String SECOND_TEST_TAG_ID = "222222";
+
+	public static final String FIRST_TEST_TAG_UID = NfcHandler.byteArrayToHex(FIRST_TEST_TAG_ID.getBytes());
+	public static final String SECOND_TEST_TAG_UID = NfcHandler.byteArrayToHex(SECOND_TEST_TAG_ID.getBytes());
+
+	public static final String THIRD_TEST_TAG_ID = "333333";
+	public static final String FOURTH_TEST_TAG_ID = "444444";
+	public static final String NFC_NDEF_STRING_1 = "I am the first!";
+	public static final String NFC_NDEF_STRING_2 = "And I am the second!";
+
+	public static final String READ_TAG_ID = "readTagId";
+	public static final String READ_TAG_MESSAGE = "readTagMessage";
+	public static final String NUM_DETECTED_TAGS = "numDetectedTags";
+
+	public static void fakeNfcTag(String uid, NdefMessage ndefMessage, Tag tag, Activity callingActivity) {
+		Class activityCls = callingActivity.getClass();
+
+		PendingIntent pendingIntent = PendingIntent.getActivity(callingActivity, 0,
+				new Intent(callingActivity, activityCls).addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP), 0);
+
+		String intentAction = NfcAdapter.ACTION_TAG_DISCOVERED;
+		byte[] tagId = uid.getBytes();
+
+		Intent intent = new Intent();
+		intent.setAction(intentAction);
+		if (tag != null) {
+			intent.putExtra(NfcAdapter.EXTRA_TAG, tag);
+		}
+		intent.putExtra(NfcAdapter.EXTRA_ID, tagId);
+		if (ndefMessage != null) {
+			intent.putExtra(NfcAdapter.EXTRA_NDEF_MESSAGES, new NdefMessage[] {ndefMessage});
+			if (NfcAdapter.ACTION_NDEF_DISCOVERED.equals(intentAction)) {
+				Uri uri = ndefMessage.getRecords()[0].toUri();
+				String mime = ndefMessage.getRecords()[0].toMimeType();
+				if (uri != null) {
+					intent.setData(uri);
+				} else {
+					intent.setType(mime);
+				}
+			}
+		}
+
+		try {
+			pendingIntent.send(callingActivity, Activity.RESULT_OK, intent);
+		} catch (PendingIntent.CanceledException e) {
+			Log.d("fakeNfcTag", e.getMessage());
+		}
+	}
+
+	public static void gotoNfcFragment(int nfcBrickPosition) {
+		onBrickAtPosition(nfcBrickPosition).onSpinner(R.id.brick_when_nfc_spinner)
+				.perform(click());
+		onView(withText(R.string.new_nfc_tag))
+				.perform(click());
+	}
+
+	public static void clickSpinnerValueOnBrick(int spinnerResourceId, int position, int stringResourceId) {
```

Comment:
lets sit together this week and refactor all those spinner utility functions out of here. (90% of those are unnecessary by now (like this one here should be already available in BrickDataInteractionWrapper, and the remaining ones we just add to wrappers)